device_service.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+)
+
+type Device struct {
+	Id       int     `json:"id"`
+	Name     string  `json:"name"`
+	Interval int     `json:"interval"`
+	Value    float32 `json:"value"`
+}
+
+type DeviceService struct {
+	Database
```

Comment:
Although this is perfectly valid in golang I think in case of using Dao in the Service embedding can be more confusing than helpful. I would just name the field.